go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/grafana/alloy
 
-go 1.22.5
+go 1.23.1
```

Comment:
Do we need to also update pyroscope here? I know there was a collision between pyroscope pprof endpoint and go 1.23?

Reply:
Run the exe and it should trigger either immediately or hitting the pprof endpoint

Reply:
The CI caught it as well it seems. Bumping `github.com/grafana/pyroscope-go/godeltaprof` to 0.1.8 will likely fix it. Here is a related fix in Grafana: https://github.com/grafana/grafana/pull/92055

Reply:
thanks, I will do it